plugins/syntax_highlighting.py

```diff
@@ -0,0 +1,90 @@
+import enum
+from pygments.formatter import Formatter
+from util import html_escape
+
+class SyntaxHighlightType(enum.Enum):
+    Bold = 'strong'
+    Italic = 'em'
+    BoldItalic = 'em2'
+    Underline = 'u'
+
+# Automatically generate the tag names
+SYNTAX_HIGHLIGHT_PREFIX = 'mathnews--code-'
+SYNTAX_HIGHLIGHT_TAGS = { member: SYNTAX_HIGHLIGHT_PREFIX + member.value for member in SyntaxHighlightType }
+
+def get_syntax_highlight_tag_name(tag_type: SyntaxHighlightType) -> str:
+    return SYNTAX_HIGHLIGHT_TAGS[tag_type]
+
+def is_highlighted(text: str) -> bool:
+    return any(tag in text for tag in SYNTAX_HIGHLIGHT_TAGS.values())
+
+class IndFormatter(Formatter):
+    """InDesign compatible formatter, based on https://pygments.org/docs/formatterdevelopment/#html-3-2-formatter
+    """
+
+    def __init__(self, **options):
+        Formatter.__init__(self, **options)
+
+        # create a dict of (start, end) tuples that wrap the
+        # value of a token so that we can use it in the format
+        # method later
+        self.styles = {}
+
+        # we iterate over the `_styles` attribute of a style item
+        # that contains the parsed style values.
+        for token, style in self.style:
+            start = end = ''
+            tag_type = None
+            # a style item is a tuple in the following form:
```

Comment:
Is this comment supposed to have something else at the end?

Reply:
I actually took that from the sample code, so I'll just remove it since it seems to be causing more confusion than it solves.